host-controller/src/main/java/org/jboss/as/host/controller/resources/StoppedServerResource.java

```diff
@@ -51,11 +59,30 @@ public void registerAttributes(ManagementResourceRegistration resourceRegistrati
 
         resourceRegistration.registerReadOnlyAttribute(ServerRootResourceDefinition.LAUNCH_TYPE, new LaunchTypeHandler(ServerEnvironment.LaunchType.DOMAIN));
         resourceRegistration.registerReadOnlyAttribute(ServerRootResourceDefinition.SERVER_STATE, (context, operation) -> {
+            readResourceServerConfig(context, operation);
+
             // note this is inconsistent with the other values, should be lower case, preserved for now.
             context.getResult().set("STOPPED");
         });
         resourceRegistration.registerReadOnlyAttribute(ServerRootResourceDefinition.RUNTIME_CONFIGURATION_STATE,
-                (context, operation) -> context.getResult().set(ClientConstants.CONTROLLER_PROCESS_STATE_STOPPED));
+                (context, operation) -> {
+                    readResourceServerConfig(context, operation);
+                    context.getResult().set(ClientConstants.CONTROLLER_PROCESS_STATE_STOPPED);
+                    }
+                );
+    }
+
+    // https://issues.jboss.org/browse/WFCORE-3338 read server-config to test name of the server in operation is present
+    private void readResourceServerConfig(OperationContext context, ModelNode operation) {
+        final PathAddress address = PathAddress.pathAddress(operation.require(OP_ADDR));
```

Comment:
This should use context.getCurrentAddress().

That's a minor thing, but I want to eliminate handlers parsing of the address from the op from the code base.

Reply:
fixed.